librsvg: add v2.56.4, v2.57.3, v2.58.2

wdconinc opened the pull request:
- [x] Needs #45729

`librsvg` v2.56.4, v2.57.3, v2.58.2, minor dependency updates per https://gitlab.gnome.org/GNOME/librsvg/-/blob/2.58.2/configure.ac?ref_type=tags, checked license, fixed url version lookup.

Test build:
```
==> Installing librsvg-2.58.2-sgqqx4gikxqzlmqdc4enwqli4b6gvpy2 [93/97]
==> No binary for librsvg-2.58.2-sgqqx4gikxqzlmqdc4enwqli4b6gvpy2 found: installing from source
==> Fetching https://download.gnome.org/sources/librsvg/2.58/librsvg-2.58.2.tar.xz
==> No patches needed for librsvg
==> librsvg: Executing phase: 'autoreconf'
==> librsvg: Executing phase: 'configure'
==> librsvg: Executing phase: 'build'
==> librsvg: Executing phase: 'install'
==> librsvg: Successfully installed librsvg-2.58.2-sgqqx4gikxqzlmqdc4enwqli4b6gvpy2
  Stage: 4.59s.  Autoreconf: 0.00s.  Configure: 3.08s.  Build: 3m 19.42s.  Install: 0.95s.  Post-install: 0.52s.  Total: 3m 28.83s
[+] /opt/software/linux-ubuntu24.04-skylake/gcc-13.2.0/librsvg-2.58.2-sgqqx4gikxqzlmqdc4enwqli4b6gvpy2
```

wdconinc committed:
librsvg: add v2.56.4, v2.57.3, v2.58.2

haampie commented on var/spack/repos/builtin/packages/librsvg/package.py:
```diff
@@ -37,16 +44,18 @@ class Librsvg(AutotoolsPackage):
-    depends_on("cairo@1.16:+gobject", when="@2.50:")
-    depends_on("cairo@1.15.12:+gobject", when="@2.44.14:")
-    depends_on("cairo@1.2.0:+gobject")
+    depends_on("cairo@1.17:+gobject+png", when="@2.57:")
+    depends_on("cairo@1.16:+gobject+png", when="@2.50:")
+    depends_on("cairo@1.15.12:+gobject+png", when="@2.44.14:")
```
```suggestion
    depends_on("cairo@1.17:", when="@2.57:")
    depends_on("cairo@1.16:", when="@2.50:")
    depends_on("cairo@1.15.12:", when="@2.44.14:")
```

haampie committed:
DRY

haampie commented:
oh, this needs a rebase now...

wdconinc commented:
Instead of fixing cairo which, as of 1.18, requires meson and doesn't have autotools anymore (https://gitlab.freedesktop.org/cairo/cairo/-/tree/1.18.0), I just removed the uninstallable version 1.18.0. That should avoid the failed build here in https://gitlab.spack.io/spack/spack/-/jobs/14210852#L450, and allow this PR to get merged.

Then we can get back to adding the cairo 1.18 versions with proper meson support (in #43586) and then adding the librsvg 2.59 versions which also switched to meson in a next PR.
